src/ripple/basics/Slice.h

```diff
@@ -64,6 +65,16 @@ class Slice
     {
     }
 
+    template <class T, class Alloc>
+    Slice(
+        std::vector<T, Alloc> const& v,
```

Comment:
Why do we need the second template parameter `Alloc` ? I couldn't find instances where we explicitly specify a non-default Allocation parameter. 

In what cases does Class-Template-Argument-Deduction (CTAD) deduce non-default values for `Alloc` ?